mass_mailing_newsletter_welcome_mail/controllers/main.py

```diff
@@ -16,16 +16,19 @@ def subscribe(self, list_id, email, **post):
         if not template:
             return result
         # Welcome new subscribers
-        contact = request.env["mail.mass_mailing.contact"].sudo().search([
-            ('list_ids', 'in', list_.ids),
-            ('email', '=', email),
-            ("opt_out", "=", False),
+        contact_obj = request.env["mail.mass_mailing.contact"].with_context(
+            default_list_ids=[list_id])
```

Comment:
Why this change?

Reply:
Can you please answer about this?

Reply:
Now in v12 the contacts are not opted out by themselves, but they are opted out or not from a particular list. One way to search contacts that are opted out from a particular list is that, by searching by the opt_oup field of mass.mailing.contact (which is now a calculated field) and passing in the context the list id.
check this
https://github.com/odoo/odoo/blob/fe35870fa0024a1e8adebb386f343f25fdea2c7b/addons/mass_mailing/models/mass_mailing.py#L250
On the other hand, I assumed that so in v11 we didn't want to send mail to contacts that had been opted out from the list, now in v12 that blacklisting exists, we don't whant either to send mail to those who are blacklisted

Reply:
Well, if they are subscribing just now, how they are going to be opted-out?

If you keep it (and for the rest of cases), please always explain this not easy to deduce things as a comment in the code.

Reply:
Maybe in Odoo 11.0 this is not necessary but in v12 yes because the re-subscription of a contact doesn't work well. I made a issue to Odoo for this: https://github.com/odoo/odoo/issues/39604

Reply:
OK, then please add that as comment:
```python
# Needed for being re-subscribed while odoo/odoo#39604 is not fixed
```